src/run-in-loop/index.ts

```diff
@@ -0,0 +1,97 @@
+import { go } from '@api3/promise-utils';
+
+import { type Logger } from '../logger';
+import { generateRandomBytes32, sleep } from '../utils';
+
+export interface RunInLoopOptions {
+  /** An API3 logger instance required to execute the callback with context. */
+  logger: Logger;
+  /** Part of every message logged by the logger. */
+  logLabel?: Lowercase<string>;
+  /**
+   * The initial delay to to wait before executing the callback for the first time. Default is 0, which means the
+   * callback is executed immediately.
+   */
+  initialDelayMs?: number;
+  /**
+   * Minimum time to wait between executions. E.g. value 500 means that the next execution will be started only after
+   * 500ms from the start of the previous one (even if the previous one ended after 150ms). Default is 0.
+   */
+  frequencyMs?: number;
+  /**
+   * Minimum time to wait between executions. E.g. value 50 means that next execution will start 50ms after the previous
+   * one ended. Default is 0.
+   */
+  minWaitTimeMs?: number;
+  /**
+   * Maximum time to wait between executions. E.g. value 100 means that the next execution will be started at most after
+   * 100ms from the end of the previous one. The maxWaitTime has higher precedence than minWaitTime of frequencyMs.
```

Comment:
"or" instead of "of" in the second sentence.

Reply:
It was supposed to be "or", but changed it to "and".